SWENT-GRP5/SOLVIT
# Integrate the providers service types to the service requests

### Description:
This pull request refactors the code to replace `ServiceRequestType` with the `Services` enum in both the service request model and UI screens. This change enhances model consistency by enabling both service requests and providers to reference the same service types, streamlining communication and reducing redundancy.

- **What issue does it solve?** This PR unifies the model by standardizing service types, eliminating the need for a separate `ServiceRequestType` type.
- **What functionality does it add?** It improves model consistency across different parts of the app (service requests and providers).

### Related Issues (if applicable):
- Fixes #142

### Changes Made:
- Replaced all instances of `ServiceRequestType` with the `Services` enum in the model and UI screens.
- Updated method signatures and data handling logic to work with the `Services` enum.
- Modified UI components that relied on `ServiceRequestType` to ensure compatibility with the new `Services` type.
- Adjusted backend communication to align with the `Services` enum.
- Updated the tests that were using the ServiceRequestType.

### How to Test:
1. Run the app make sure that the services options appear when creating and editing a service request.
2. Run the androidTests for the service requests.

---
### Checklist:
- [x] My code follows the coding standards for this project
- [x] I have written new tests and/or updated existing tests
- [x] I have manually tested the functionality
- [x] The code is free of any known bugs
